src/coreclr/jit/emitarm64.h

```diff
@@ -967,7 +967,14 @@ inline static bool insOptsScalable(insOpts opt)
 {
     // `opt` is any of the scalable types.
     return ((opt == INS_OPTS_SCALABLE_B) || (opt == INS_OPTS_SCALABLE_H) || (opt == INS_OPTS_SCALABLE_S) ||
-            (opt == INS_OPTS_SCALABLE_D) ||  (opt == INS_OPTS_SCALABLE_Q));
+            (opt == INS_OPTS_SCALABLE_D) || (opt == INS_OPTS_SCALABLE_Q));
+}
+
+inline static bool insOptsScalableStandard(insOpts opt)
```

Comment:
wondering why we need a different method for `INS_OPTS_SCALABLE_Q`?

Reply:
Most instructions don't allow Q sized registers. It's only a small few.

I think `insOptsScalable()` should still check for _everything_, even though that's not the default.

Prior to this PR, most instructions were using `insOptsScalableSimple()` but that vanished when I got rid of all the enum entries. For simplicity, I could rename `insOptsScalableStandard()` to `insOptsScalableSimple()` and that'd reduce the diff in this PR.

Reply:
I am thinking why not just have `insOptsScalableStandard()` (or `insOptsScalableSimple()`). Just have 1 method which is `insOptsScalable()` that takes care of `_Q` as well.

Reply:
That will mean for most instructions we will allow `Q` to be passed as a size.

Which would then probably assert when trying to encode the instruction, in `insEncodeElemsize()` (because it doesn't support `Q`)

Reply:
If an instruction wrongly passes `_Q` where it is not supposed to, `insEncodeElemsize()` will correctly assert, right? I am trying to understand why `_Q` needs special treatment than others.

Reply:
Only reason it would need special treatment is that we already have the various size checks for other instructions. `insOptsScalableWide()` is used for widening instructions (B,H,S only), `insOptsScalableFloat()` is used for float types (H,S,D), `insOptsScalableWords()` is just S and D, etc etc. Those are used throughout `emitInsSanityCheck()`, instead of using `insOptsScalable()`. It would seem odd to check for restricted element sizes for some instructions, and then for others allow Q (via use of `insOptsScalable()`).

Reply:
I see, alright that makes sense then.

Reply:
nit for next PR: we can rewrite it by:

```c++
inline static bool insOptsScalable(insOpts opt)
{
    // `opt` is any of the scalable types.
    return (insOptsScalableStandard(opt) || (opt == INS_OPTS_SCALABLE_Q));
}
```